packages/thirdweb/src/pay/buyWithCrypto/actions/getQuote.ts

```diff
@@ -0,0 +1,316 @@
+
+    fromAmount: string;
+    fromAmountWei: string;
+
+    toAmountMinWei: string;
+    toAmountMin: string;
+    toAmount: string;
+    toAmountWei: string;
+
+    estimated: {
+      fromAmountUSDCents: number;
+      toAmountMinUSDCents: number;
+      toAmountUSDCents: number;
+      slippageBPS: number;
+      feesUSDCents: number;
+      gasCostUSDCents?: number;
+      durationSeconds?: number;
+    };
+
+    maxSlippageBPS: number;
+  };
+
+  paymentTokens: QuotePaymentToken[];
+  processingFees: QuotePaymentToken[];
+  client: ThirdwebClient;
+};
+
+/**
+ * Get a quote of type [`BuyWithCryptoQuote`](https://portal.thirdweb.com/references/typescript/v5/BuyWithCryptoQuote) to buy any given token with crypto.
+ * This quote contains the information about the swap such as token amounts, processing fees, estimated time etc.
+ *
+ * Once you have the quote, you can use `prepareTransaction` and prepare the transaction for submission.
+ * @param params - object of type [`GetBuyWithCryptoQuoteParams`](https://portal.thirdweb.com/references/typescript/v5/GetBuyWithCryptoQuoteParams)
+ * @returns Object of type [`BuyWithCryptoQuote`](https://portal.thirdweb.com/references/typescript/v5/BuyWithCryptoQuote) which contains the information about the quote such as processing fees, estimated time, converted token amounts, etc.
+ * @example
+ *
+ * ```ts
+ * import { getBuyWithCryptoQuote } from "thirdweb/pay";
+ *
+ * const quote = await getBuyWithCryptoQuote({
+ *  client,
+ *  fromAddress: "0x...", // wallet address
+ *  fromChainId: 137, // chain id of the source token
+ *  fromTokenAddress: "0x...", // token address of the source token
+ *  fromAmount: "10", // amount of source token to swap
+ *  // optionally, you can use `toAmount` instead if you only want a certain amount of destination token
+ *  toChainId: 10, // chain id of the destination token
+ *  toTokenAddress: "0x...", // token address of the destination token
+ *  toAddress: "0x...", // optional: send the tokens to a different address
+ *  maxSlippageBPS: 50, // optional: max 0.5% slippage
+ * });
+ * ```
+ */
+export async function getBuyWithCryptoQuote(
+  params: getBuyWithCryptoQuoteParams,
+): Promise<BuyWithCryptoQuote> {
+  try {
+    const urlParamsObj: Record<string, string> = {
+      fromAddress: params.fromAddress,
+      fromChainId: params.fromChainId.toString(),
+      fromTokenAddress: params.fromTokenAddress.toLowerCase(),
+      toChainId: params.toChainId.toString(),
+      toTokenAddress: params.toTokenAddress.toLowerCase(),
+    };
+
+    if ("fromAmount" in params && params.fromAmount) {
+      urlParamsObj.fromAmount = params.fromAmount;
+    }
+
+    if ("toAmount" in params && params.toAmount) {
+      urlParamsObj.toAmount = params.toAmount;
+    }
+
+    if (params.maxSlippageBPS) {
+      urlParamsObj.maxSlippageBPS = params.maxSlippageBPS.toString();
+    }
+
+    const queryString = new URLSearchParams(urlParamsObj).toString();
```

Comment:
same thing, please just construct this directly onto the querystring and initialize the `new URLSearchParams()` up top